File: test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/StatechartLocalReactionsTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for StatechartLocalReactions
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class StatechartLocalReactionsTest extends AbstractExecutionFlowTest {
	@Inject
	private SCTUnitTestModels models;
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("StatechartLocalReactions.sct");
		initInterpreter(flow);
	}
	@Test
	public void statechartLocalReactionsTest() throws Exception {
		interpreter.enter();
		assertTrue(isActive("S1"));
		assertTrue(isActive("a"));
		while (getInteger("myInt") < 10) {
			assertTrue(isActive("a"));
			if (getInteger("myInt") % 2 == 0) {
				assertTrue(isActive("S1"));;
			} else {
				assertTrue(isActive("S2"));;
			}
			interpreter.runCycle();
		}
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/StatechartLocalReactionsTest/StatechartLocalReactionsTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "StatechartLocalReactions.h"

StatechartLocalReactions handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechartLocalReactions_init(&handle);
	}
};


TEST_F(StatemachineTest, statechartLocalReactionsTest) {					
	statechartLocalReactions_enter(&handle);
	EXPECT_TRUE(statechartLocalReactions_isStateActive(&handle, StatechartLocalReactions_main_region_S1));
	EXPECT_TRUE(statechartLocalReactions_isStateActive(&handle, StatechartLocalReactions_region2_a));
	while (statechartLocalReactionsIface_get_myInt(&handle)< 10l) {
		EXPECT_TRUE(statechartLocalReactions_isStateActive(&handle, StatechartLocalReactions_region2_a));
		if (statechartLocalReactionsIface_get_myInt(&handle)%2l== 0l) {
			EXPECT_TRUE(statechartLocalReactions_isStateActive(&handle, StatechartLocalReactions_main_region_S1));
		}
		else {
			EXPECT_TRUE(statechartLocalReactions_isStateActive(&handle, StatechartLocalReactions_main_region_S2));
		}
		statechartLocalReactions_runCycle(&handle);
	}
}


